runtime/src/bank.rs

```diff
@@ -638,6 +647,16 @@ impl Bank {
         self.slots_per_epoch
     }
 
+    /// Return the checkpointed bank that should be used to generate a leader schedule.
+    /// Return None if a sufficiently old bank checkpoint doesn't exist.
+    pub fn leader_schedule_bank(&self) -> Option<Arc<Bank>> {
+        let epoch_slot_height = self.slot_height() - self.slot_index();
+        let expected = epoch_slot_height.saturating_sub(self.leader_schedule_slot_offset);
```

Comment:
By the time you're looking for this, this bank should have already been merged into root and finalized, no?

Reply:
Yeah. Per our conversation, we need to make our "merge into root" function aware that we're looking back in time to query for leader schedules.

Reply:
Yeah it seems that merge into root should know when its about to merge a slot crosses an epoch boundary and needs to be used for leader schedule generation

Reply:
that logic should be in replay_stage or locktower, the code that would call bank.merge_parents()